src/module-elasticsuite-catalog/etc/di.xml

```diff
@@ -330,4 +330,41 @@
     <type name="\Magento\Catalog\Api\Data\ProductAttributeInterface">
         <plugin name="resetIndexConfigAfterAttributeSave" type="\Smile\ElasticsuiteCatalog\Plugin\Catalog\Eav\AttributePlugin" />
     </type>
+
+    <type name="Smile\ElasticsuiteCore\Model\Search\Request\Field\Mapper">
+        <arguments>
+            <argument name="fieldNameMapping" xsi:type="array">
+                <item name="price" xsi:type="string">price.price</item>
+                <item name="position" xsi:type="string">category.position</item>
+                <item name="category_ids" xsi:type="string">category.category_id</item>
+            </argument>
+        </arguments>
+    </type>
+
+    <type name="Smile\ElasticsuiteCatalogRule\Model\Rule\Condition\Product\AttributeList">
+        <arguments>
+            <argument name="requestFieldMapper" xsi:type="object">Smile\ElasticsuiteCatalog\Model\Layer\Filter\Field\Mapper</argument>
+        </arguments>
+    </type>
+    <virtualType name="Smile\ElasticsuiteCatalog\Model\Layer\Filter\Field\Mapper" type="Smile\ElasticsuiteCore\Model\Search\Request\Field\Mapper">
+        <arguments>
+            <argument name="fieldNameMapping" xsi:type="array">
+                <item name="price" xsi:type="string">price.price</item>
+                <item name="categories" xsi:type="string">category.category_id</item>
```

Comment:
Maybe the special case about "categories" should be injected by the virtual-category module, since this field naming is introduced by `Smile\ElasticsuiteVirtualCategory\Model\Layer\Filter\Category`

Reply:
Actually the "categories" mapping is not needed at all. Further more the method `\Smile\ElasticsuiteCatalog\Model\Layer\Filter\Category::getFilterField` is not used and can not be used in neither category legacy filter or virtual category filter. 

The reason it can not be used is because
`\Smile\ElasticsuiteCatalog\Search\Request\Product\Attribute\Aggregation\Category::getAggregationData`
and
`\Smile\ElasticsuiteVirtualCategory\Search\Request\Product\Attribute\Aggregation\Category::getAggregationData`

are using custom bucket naming for categories bucket which is not consistent with the other filters, for example:
`\Smile\ElasticsuiteCatalog\Search\Request\Product\Attribute\Aggregation\Price::getAggregationData`